pkg/migrate/cstor/pool.go

```diff
@@ -466,13 +530,28 @@ func (c *CSPCMigrator) updateCVRsLabels(cspObj *apis.CStorPool, cspiObj *cstor.C
 
 func addSkipAnnotationToSPC(spcObj *apis.StoragePoolClaim) error {
 retry:
-	spcObj.Annotations = map[string]string{
-		"openebs.io/skip-validations": "true",
+	if spcObj.Annotations == nil {
+		spcObj.Annotations = map[string]string{}
 	}
+	spcObj.Annotations["openebs.io/skip-validations"] = "true"
 	_, err := spc.NewKubeClient().Update(spcObj)
 	if k8serrors.IsConflict(err) {
 		klog.Errorf("failed to update spc with skip-validation annotation due to conflict error")
 		goto retry
 	}
 	return err
 }
+
+func addCSPCAnnotationToSPC(spcObj *apis.StoragePoolClaim, cspcName string) error {
+retry:
+	if spcObj.Annotations == nil {
+		spcObj.Annotations = map[string]string{}
+	}
+	spcObj.Annotations[types.CStorPoolClusterLabelKey] = cspcName
+	_, err := spc.NewKubeClient().Update(spcObj)
+	if k8serrors.IsConflict(err) {
+		klog.Errorf("failed to update spc with cspc annotation due to conflict error")
```

Comment:
add sleep of 2 or 5 seconds

Reply:
The sleep can be equal to sync interval which can end in a cyclic loop of conflict errors